packages/mui-base/src/SliderUnstyled/useSlider.tsx

```diff
@@ -0,0 +1,668 @@
+    }
+    const finger = trackFinger(nativeEvent, touchId);
+    if (finger !== false) {
+      const { newValue, activeIndex } = getFingerNewValue({ finger, values });
+      focusThumb({ sliderRef, activeIndex, setActive });
+
+      setValueState(newValue);
+
+      if (handleChange) {
+        handleChange(nativeEvent, newValue, activeIndex);
+      }
+    }
+
+    moveCount.current = 0;
+    const doc = ownerDocument(sliderRef.current);
+    doc.addEventListener('touchmove', handleTouchMove);
+    doc.addEventListener('touchend', handleTouchEnd);
+  });
+
+  const stopListening = React.useCallback(() => {
+    const doc = ownerDocument(sliderRef.current);
+    doc.removeEventListener('mousemove', handleTouchMove);
+    doc.removeEventListener('mouseup', handleTouchEnd);
+    doc.removeEventListener('touchmove', handleTouchMove);
+    doc.removeEventListener('touchend', handleTouchEnd);
+  }, [handleTouchEnd, handleTouchMove]);
+
+  React.useEffect(() => {
+    const { current: slider } = sliderRef;
+    slider!.addEventListener('touchstart', handleTouchStart, {
+      passive: doesSupportTouchActionNone(),
+    });
+
+    return () => {
+      // @ts-ignore
+      slider!.removeEventListener('touchstart', handleTouchStart, {
+        passive: doesSupportTouchActionNone(),
+      });
+
+      stopListening();
+    };
+  }, [stopListening, handleTouchStart]);
+
+  React.useEffect(() => {
+    if (disabled) {
+      stopListening();
+    }
+  }, [disabled, stopListening]);
+
+  const handleMouseDown = useEventCallback(
+    (event: React.MouseEvent<HTMLSpanElement, MouseEvent>) => {
+      if (onMouseDown) {
+        onMouseDown(event);
+      }
+
+      // Only handle left clicks
+      if (event.button !== 0) {
+        return;
+      }
+
+      // Avoid text selection
+      event.preventDefault();
+      const finger = trackFinger(event, touchId);
+      if (finger !== false) {
+        const { newValue, activeIndex } = getFingerNewValue({ finger, values });
+        focusThumb({ sliderRef, activeIndex, setActive });
+
+        setValueState(newValue);
+
+        if (handleChange) {
+          handleChange(event, newValue, activeIndex);
+        }
+      }
+
+      moveCount.current = 0;
+      const doc = ownerDocument(sliderRef.current);
+      doc.addEventListener('mousemove', handleTouchMove);
+      doc.addEventListener('mouseup', handleTouchEnd);
+    },
+  );
+
+  const trackOffset = valueToPercent(range ? values[0] : min, min, max);
+  const trackLeap = valueToPercent(values[values.length - 1], min, max) - trackOffset;
+  const trackStyle = {
+    // @ts-ignore
+    ...axisProps[axis].offset(trackOffset),
+    // @ts-ignore
+    ...axisProps[axis].leap(trackLeap),
+  };
+
+  const getRootProps = () => {
+    return {
+      ref: handleRef,
+      onMouseDown: handleMouseDown,
+    };
+  };
+
+  const getTrackProps = () => {
+    return {
+      style: { ...trackStyle, ...componentsProps.track?.style },
```

Comment:
Yeah, let me try to refactor and see where we'll land

Reply:
This would create a problem I think. I would expect that the unstyled version should handle this. If we let this be a responsibility of the users of the hook/unstyled component, we would need to document how each developer should set it, as it is related to the functionality of the component.

I will however move it to the `SliderUnstyled` as it is already setting `style` on some places for the `thumbs`, `valueLabel` etc, so that at least those will be on one place (in the component).